base/windows/shop/ShopCharDisplay.js

```diff
@@ -0,0 +1,273 @@
+        this.up_arrow.alpha = 0;
+        this.down_arrow.alpha = 0;
+
+        this.arrow_tweens = [];
+
+        this.lines = [];
+        this.current_line = 0;
+        this.selected_index = 0;
+        this.is_active = false;
+        this.is_open = false;
+    }
+
+    update_group_pos(){
+        this.char_group.x = CHAR_GROUP_X - SHIFT_X + this.game.camera.x;
+        this.char_group.y = CHAR_GROUP_Y - SHIFT_Y + this.game.camera.y;
+        this.arrow_group.x = ARROW_GROUP_X + this.game.camera.x;
+        this.arrow_group.y = ARROW_GROUP_Y + this.game.camera.y;
+    }
+
+    /*Hides or shows specific arrows
+    
+    Input: up, down [boolean] - If true, shows up/down arrow*/
+    set_arrows(up=false, down=false){
+        this.up_arrow.x = UP_ARROW_X;
+        this.up_arrow.y = UP_ARROW_Y;
+        this.down_arrow.x = DOWN_ARROW_X;
+        this.down_arrow.y = DOWN_ARROW_Y;
+        if(up) this.up_arrow.alpha = 1;
+        else this.up_arrow.alpha = 0;
+
+        if(down) this.down_arrow.alpha = 1;
+        else this.down_arrow.alpha = 0;
+    }
+
+    /*Checks which arrows to show or hide*/
+    check_arrows(){
+        let up = false;
+        let down = false;
+
+        if(this.current_line < this.lines.length-1) down = true;
+        if(this.current_line > 0) up = true;
+
+        this.set_arrows(up, down);
+        this.init_arrow_tweens();
+        this.game.world.bringToTop(this.arrow_group);
+    }
+
+    /*Starts the arrow animations*/
+    init_arrow_tweens(){
+        let up_tween = this.game.add.tween(this.up_arrow)
+                .to({y: UP_ARROW_Y - ARROW_Y_DIFF}, ARROW_TWEEN_TIME, Phaser.Easing.Linear.None)
+                .to({y: UP_ARROW_Y}, ARROW_TWEEN_TIME, Phaser.Easing.Linear.None).loop();
+        this.arrow_tweens.push(up_tween);
+
+        let down_tween = this.game.add.tween(this.down_arrow)
+                .to({y: DOWN_ARROW_Y + ARROW_Y_DIFF}, ARROW_TWEEN_TIME, Phaser.Easing.Linear.None)
+                .to({y: DOWN_ARROW_Y}, ARROW_TWEEN_TIME, Phaser.Easing.Linear.None).loop();
+        this.arrow_tweens.push(down_tween);
+
+        up_tween.start();
+        down_tween.start();
+    }
+
+    /*Clears the arrow animations*/
+    clear_arrow_tweens(){
+        for(let i=0; i<this.arrow_tweens.length; i++){
+            this.game.tweens.remove(this.arrow_tweens.pop());
+        }
+    }
+
+    set_chars() {
+        for (let i = 0; i < this.lines[this.current_line].length; ++i) {
+            let char = this.lines[this.current_line][i];
+            let sprite = null;
+
+            let dead_idle = this.char_group.children.filter(s => { return (s.alive === false && s.key === char.key_name + "_idle"); });
+
+            if(dead_idle.length>0) sprite = dead_idle[0].reset(i*GAP_SIZE, 0);
+            else sprite = this.char_group.create(i*GAP_SIZE, 0, char.key_name + "_idle");
+
+            char.sprite_base.setAnimation(sprite, "idle");
```

Comment:
Consider using `base_actions.IDLE` from utils.js instead of writing "idle" literal.